### ISSUE TITLE
Wasserstein Training

### ISSUE THREAD

**Issue by loparcog** [McMasterAI/StyleGAN#22]
# What is it

Wasserstein GANs are essentially an addition to regular GANs, providing a new way to train models that improves stability and possibly image output. A good article on the topic can be found [at the machinelearningmastery website](https://machinelearningmastery.com/how-to-implement-wasserstein-loss-for-generative-adversarial-networks/).

# What does it change

Implementing WGAN features into our base GAN would involve changing the loss function and possibly some training settings.

# Goals to see

Hopefully this would produce better and more consistent/stable results when training, meaning the loss functions are steadily reducing rather than bouncing or increasing for substantial amounts of time
